art/attacks/attack.py

```diff
@@ -45,15 +45,15 @@ def __init__(self, classifier):
         """
         self.classifier = classifier
 
-    def generate(self, x, **kwargs):
+    def generate(self, x, y):
         """
         Generate adversarial examples and return them as an array. This method should be overridden by all concrete
         attack implementations.
 
         :param x: An array with the original inputs to be attacked.
         :type x: `np.ndarray`
-        :param kwargs: Attack-specific parameters used by child classes.
-        :type kwargs: `dict`
+        :param y: An array with the original labels to be predicted.
```

Comment:
The doctring for `y` is unclear. Here is a suggestion for the base class `Attack`:
```suggestion
        :param y: Correct labels or target labels for `x`, depending if the attack is targeted
               or not. This parameter is only used by some of the attacks.
```
For the other attacks where you have used the same string, you probably should make it explicit how `y` is used depending if the attack is targeted or not (you can reuse the docstring from `carlini.py`), or just state that the respective attack ignores this parameter.